packages/dev-server/src/HermesBundler.ts

```diff
@@ -0,0 +1,154 @@
+import type { Platform } from '@expo/config';
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs-extra';
+import type { composeSourceMaps } from 'metro-source-map';
+import os from 'os';
+import path from 'path';
+import process from 'process';
+import resolveFrom from 'resolve-from';
+
+export async function shouldBuildHermesBundleAsync(
+  projectRoot: string,
+  platform: Platform
+): Promise<boolean> {
+  if (platform === 'android') {
+    const gradlePropertiesPath = path.join(projectRoot, 'android', 'gradle.properties');
+    if (!fs.existsSync(gradlePropertiesPath)) {
+      // TODO(kudo): Clarify default behavior in managed workflow for gradle.properties not exist
+      return false;
+    }
+    const properties = parseGradleProperties(await fs.readFile(gradlePropertiesPath, 'utf8'));
+    for (const [key, value] of properties) {
+      if (key === 'JS_RUNTIME' && value === 'hermes') {
```

Comment:
This seems interesting. In gradle properties, the format is often like `foo.bar=true`, but we've opted for this env-variable type format `FOO_BAR=true`. If this logic is to stay here, we'll need to document that we require this value to be defined in order for hermes to work, we'd probably also need to spec which format ppl should use in their gradle.properties files.

Reply:
that is really interesting. we might [doing wrong for years](https://github.com/facebook/react-native/blob/master/ReactAndroid/gradle.properties).

also having a side question for managed workflow.
assuming gradle.properties is not existed in managed workflow project, where do we specify the js runtime?
perhaps we should moving this setting directly into app.json?

Reply:
i think we probably want to add a property to app.json and app.config.js, yeah. so, possibly something like:

```json
{
  "expo" {
     "slug": "abcd",
     "engine": "hermes"
  }
}
```

one consideration here is that if we don't end up supporting hermes in expo go then this field may be a bit confusing.